com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/cf/util/DeployedLiveModuleCalculatorTest.java

```diff
@@ -0,0 +1,94 @@
+package com.sap.cloud.lm.sl.cf.core.cf.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.sap.cloud.lm.sl.cf.core.model.ApplicationColor;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.common.util.MapUtil;
+
+public class DeployedLiveModuleCalculatorTest {
+
+    private static Stream<Arguments> testGetLiveDeployedModule() {
+        return Stream.of(
+        // @formatter:off
+               // (1) Get module-1 from deployed modules [module-1, module-2]
+               Arguments.of(Arrays.asList(MapUtil.asMap("module-1", "foo-blue"), MapUtil.asMap("module-2", "bar-green")), ApplicationColor.BLUE, MapUtil.asMap("module-1", "foo-blue")),
+               // (2) Get live application in case where both blue and green are deployed
+               Arguments.of(Arrays.asList(MapUtil.asMap("module-1", "foo-blue"), MapUtil.asMap("module-1", "foo-green")), ApplicationColor.GREEN, MapUtil.asMap("module-1", "foo-green")),
+               // (3) Get module-2 from deployed modules [module-1, module-2] without colors
+               Arguments.of(Arrays.asList(MapUtil.asMap("module-1", "foo"), MapUtil.asMap("module-2", "bar")), null, MapUtil.asMap("module-2", "bar")),
+               // (4) Test without deployed mta
+               Arguments.of(Collections.emptyList(), null, Collections.emptyMap()),
+               // (5) Get application without color suffix when previous deployed color is BLUE
+               Arguments.of(Arrays.asList(MapUtil.asMap("module-1", "foo"), MapUtil.asMap("module-1", "foo-green")), ApplicationColor.BLUE, MapUtil.asMap("module-1", "foo")),
+               // (6) Get live application (foo-green) when foo and foo-blue exists
+               Arguments.of(Arrays.asList(MapUtil.asMap("module-1", "foo"), MapUtil.asMap("module-1", "foo-blue"), MapUtil.asMap("module-1", "foo-green")), ApplicationColor.GREEN, 
+                            MapUtil.asMap("module-1", "foo-green"))
+        // @formatter:on
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void testGetLiveDeployedModule(List<Map<String, String>> deployedModulesNames, ApplicationColor deployedMtaColor,
+                                          Map<String, String> expectedLiveDeployedModuleName) {
```

Comment:
Why do you need map here(expectedLiveDeployedModuleName), only the module names?